internal/js/modules/k6/browser/common/page.go

```diff
@@ -1826,6 +1830,41 @@ func (p *Page) WaitForURL(urlPattern string, opts *FrameWaitForURLOptions, jsReg
 	return nil
 }
 
+// WaitForResponse waits for a response that matches the given URL pattern.
+// jsRegexChecker should be non-nil to be able to test against a URL pattern.
+func (p *Page) WaitForResponse(
+	urlOrRegex string, opts *PageWaitForResponseOptions, jsRegexChecker JSRegexChecker,
+) (*Response, error) {
+	p.logger.Debugf("Page:WaitForResponse", "sid:%v pattern:%s", p.sessionID(), urlOrRegex)
+	_, span := TraceAPICall(p.ctx, p.targetID.String(), "page.waitForResponse")
+	defer span.End()
+
+	timeoutCtx, timeoutCancel := context.WithTimeout(p.ctx, opts.Timeout)
+	defer timeoutCancel()
+
+	// Create URL matcher based on the pattern
+	matcher, err := urlMatcher(urlOrRegex, jsRegexChecker)
+	if err != nil {
+		spanRecordError(span, err)
+		return nil, fmt.Errorf("parsing URL pattern: %w", err)
+	}
+
+	// Use the dedicated ResponseEventHandler for clean separation
+	resp, err := p.responseEventHandler.waitForMatch(timeoutCtx, matcher)
+	if err != nil {
+		if errors.Is(err, context.DeadlineExceeded) {
+			e := &k6ext.UserFriendlyError{
+				Err:     err,
+				Timeout: opts.Timeout,
+			}
+			err = fmt.Errorf("timed out waiting for response - %q: %w", urlOrRegex, e)
```

Comment:
I'm not 100% sure about this. The context could close because the:

1. the iteration is ending;
2. the test is ending;
3. or when the timeout is hit.